ast/src/analyzed/mod.rs

```diff
@@ -107,6 +107,34 @@ impl<T> Analyzed<T> {
             + 1
     }
 
+    pub fn next_id_for_kind(&self, poly_type: PolynomialType) -> u64 {
+        match poly_type {
+            PolynomialType::Committed => {
+                self.committed_polys_in_source_order()
+                    .map(|(symbol, _)| symbol.id)
+                    .max()
+                    .unwrap_or(0)
+                    + 1
+            }
+            PolynomialType::Constant => {
+                self.constant_polys_in_source_order()
+                    .map(|(symbol, _)| symbol.id)
+                    .max()
```

Comment:
Can the `.max().unwrap_or(0) + 1` be shared across the branches, or at least part of it?

Reply:
Done